packages/components/src/components/modal/_modal.scss

```diff
@@ -83,31 +84,98 @@
     transition: transform $duration--moderate-02 motion(exit, expressive);
 
     @include carbon--breakpoint(md) {
-      width: 50%;
+      width: 84%;
       max-width: 768px;
       max-height: 90%;
       height: auto;
     }
 
     @include carbon--breakpoint(lg) {
-      max-height: 80%;
+      width: 60%;
+      max-height: 84%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 48%;
+    }
+  }
+
+  .#{$prefix}--modal-container--xs {
+    @include carbon--breakpoint(md) {
+      width: 48%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 32%;
+      max-height: 48%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 24%;
+
+      .#{$prefix}--modal-header,
+      .#{$prefix}--modal-content,
+      .#{$prefix}--modal-content__regular-content {
+        padding-right: rem(16px);
+      }
+    }
+  }
+
+  .#{$prefix}--modal-container--sm {
+    @include carbon--breakpoint(md) {
+      width: 60%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 42%;
+      max-height: 72%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 36%;
+    }
+  }
+
+  .#{$prefix}--modal-container--lg {
+    @include carbon--breakpoint(md) {
+      width: 96%;
+    }
+
+    @include carbon--breakpoint(lg) {
+      width: 84%;
+      max-height: 96%;
+    }
+
+    @include carbon--breakpoint(xlg) {
+      width: 72%;
     }
   }
 
   .#{$prefix}--modal-header,
   .#{$prefix}--modal-content {
-    padding-right: 25%;
     padding-left: 1rem;
   }
 
   .#{$prefix}--modal-header,
-  .#{$prefix}--modal-footer {
-    flex-shrink: 0;
+  .#{$prefix}--modal-content,
+  .#{$prefix}--modal-content__regular-content {
+    padding-right: 20%;
+  }
+
+  .#{$prefix}--modal-content--with-form {
+    padding-right: 1rem;
```

Comment:
should we use a spacing token here instead?

Reply:
Seems that the modal content historically uses fixed size: https://github.com/carbon-design-system/carbon/blob/v10.7.0/packages/components/src/components/modal/_modal.scss#L100

@aagonzales Do you think it's token-based instead?

Reply:
Margin-right is the special one because sometimes its 20%. I leave it up to dev to decide if its better to use `1rem` or `spacing-05`.

Reply:
Thanks @aagonzales for your response - Updated with spacing tokens.